### PR TITLE
requesthandler: Add optional context to TriggerHotkeyByName

### PR DESCRIPTION
**Pull request by exeldro**
### Description
Add optional context to TriggerHotkeyByName

### Motivation and Context
The same hotkey name can exist on multiple context, to make sure we got the correct one add an optional context

### How Has This Been Tested?
Tried calling 
`{
  "requestType": "TriggerHotkeyByName",
  "requestData": {
    "hotkeyName": "OBSBasic.SelectScene",
    "contextName": "Scene 1"
  }
}`
Tested OS(s): Windows 11

### Types of changes
 - New request/event (non-breaking) 
 
### Checklist:
-  [x] I have read the [Contributing Guidelines](https://github.com/obsproject/obs-websocket/wiki/Contributing-Guidelines).
-  [x] All commit messages are properly formatted and commits squashed where appropriate.
-  [x] My code is not on `master` or a `release/*` branch.
-  [x] The code has been tested.
-  [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: src/requesthandler/RequestHandler_General.cpp [hunk 1]
```diff
@@ -248,7 +249,8 @@ RequestResult RequestHandler::TriggerHotkeyByName(const Request &request)
 	if (!request.ValidateString("hotkeyName", statusCode, comment))
 		return RequestResult::Error(statusCode, comment);
 
-	obs_hotkey_t *hotkey = Utils::Obs::SearchHelper::GetHotkeyByName(request.RequestData["hotkeyName"]);
+	obs_hotkey_t *hotkey = Utils::Obs::SearchHelper::GetHotkeyByName(
+		request.RequestData["hotkeyName"], request.Contains("contextName") ? request.RequestData["contextName"] : "");
```
Comment (tt2468) [inline src/requesthandler/RequestHandler_General.cpp:253]: To maintain code consistency, please implement the optional context name field like done here: https://github.com/obsproject/obs-websocket/blob/ee2e24e42f36faedbca486c7a308444c0f3c3fc7/src/requesthandler/RequestHandler_Inputs.cpp#L40

(I understand it seems a bit wordy, but I would rather change every similar instance to something else in the future than have a bunch of different implementations of the same logic sequence)

File: src/utils/Obs_SearchHelper.cpp [hunk 2]
```diff
@@ -19,16 +19,58 @@ with this program. If not, see <https://www.gnu.org/licenses/>
...
 
 	auto hotkeys = ArrayHelper::GetHotkeyList();
 
 	for (auto hotkey : hotkeys) {
-		if (obs_hotkey_get_name(hotkey) == name)
+		if (obs_hotkey_get_name(hotkey) != name)
+			continue;
+		if (context.empty())
 			return hotkey;
+		auto type = obs_hotkey_get_registerer_type(hotkey);
```
Comment (tt2468) [inline src/utils/Obs_SearchHelper.cpp:34]: Nitpick: Add a newlines to isolate the if statements a bit from each other.

File: src/utils/Obs_SearchHelper.cpp [hunk 3]
```diff
@@ -19,16 +19,58 @@ with this program. If not, see <https://www.gnu.org/licenses/>
...
+		if (type == OBS_HOTKEY_REGISTERER_FRONTEND) {
+		} else if (type == OBS_HOTKEY_REGISTERER_SOURCE) {
+			auto source = obs_weak_source_get_source((obs_weak_source_t *)obs_hotkey_get_registerer(hotkey));
```
Comment (tt2468) [inline src/utils/Obs_SearchHelper.cpp:37]: Please use `OBS[*]AutoRelease` instead of manual releases

File: src/utils/Obs_SearchHelper.cpp [hunk 4]
```diff
@@ -19,16 +19,58 @@ with this program. If not, see <https://www.gnu.org/licenses/>
...
 	auto hotkeys = ArrayHelper::GetHotkeyList();
 
 	for (auto hotkey : hotkeys) {
-		if (obs_hotkey_get_name(hotkey) == name)
+		if (obs_hotkey_get_name(hotkey) != name)
+			continue;
+		if (context.empty())
 			return hotkey;
+		auto type = obs_hotkey_get_registerer_type(hotkey);
+		if (type == OBS_HOTKEY_REGISTERER_FRONTEND) {
```
Comment (tt2468) [inline src/utils/Obs_SearchHelper.cpp:35]: I don't think this does anything?

**Review by tt2468 — CHANGES_REQUESTED** [review]

File: src/utils/Obs_SearchHelper.cpp (hunk 4 above)
Comment (exeldro) [reply src/utils/Obs_SearchHelper.cpp]: removed
